### ISSUE TITLE
How can scaled Actors participate in outer container layout such that they conform to prejudices?

### ISSUE THREAD

**Issue by SomeTroglodyte** [libgdx/libgdx#7223]
#### Issue details
Using Scene2d Tables and Labels, I am trying to solve the following half-abstract puzzle:
* Fill a Table with Label(s) and stuff, expecting dynamic content and therefore dynamic width.
* If the prefWidth now exceeds stage width, turn on isTransform and setScale(stage.width / prefWidth)
* Example: Table says it wants 800 units, stage says 500 units. Scale 0.625 should allow it to fit, right? Also, prefHeight was 34, so now it should only need 22 vertically.
* Task: insert that scaled Table into a container Table (or any other construct) such that the container knows our candidate wants 500x22 units and does layout exactly as if it were an unscaled Table needing 500x22 units.

I tried, tried, tried and failed. So far my best intermediate wrapper would still misalign the candidate, move it out of the bounds of the container Cell, and report contradictory heights: I got, inspecting the outmost containing Cell, an actorX == -250 (-150 I would understand: centered using unscaled width, no idea where the extra -100 could come from), actorWidth 500, actorHeight 34, and on the outer Table prefHeight != sum(rowPrefHeight)...

I'm looking not for in-depth analysis or instructions, just a general guideline - how would you approach the task set out above?

No sample code - I discarded all experiments so far, from "closer but still off" I concluded "wrong track entirely".

And yes, I know a workaround would be to not use scale at all but project paddings, font sizes, and icon dimensions separately.

#### Version of libGDX and/or relevant dependencies
1.11.0

#### Please select the affected platforms
- [ ] Android
- [ ] iOS
- [ ] HTML/GWT
- [x] Windows
- [x] Linux
- [ ] macOS

Little aside: On https://libgdx.com/wiki/articles/getting-help#help-us-help-you, there's a link "post a new issue" just above that anchor that gives an error.

**Comment by NathanSweet** [comment]
It's a littler hard to follow but I think I get it. The table doesn't take into account scale when computing pref width. So, fill your table, get the pref width, set the table's scale as needed. Get that working in simple-as-possible example code where you draw the table as an actor at 0,0 (not put into any layout).

Next you need the container. Extend WidgetGroup and implement the logic to return the pref width, etc that you want. Add the table as a child actor at 0,0.

**Comment by NathanSweet** [comment]
This is an issue tracker, so not the best place for questions. You can get help more quickly on [Discord](https://libgdx.com/community/discord/). We can continue this discussion here, but I'll close this issue.

**Comment by NathanSweet** [comment]
Hope it helped! Most layouting widgets don't account for scale, so putting the table in a layout probably will not do what you want (the layout would see the table's pref width, not the lesser, scaled width). You need an actor that knows about the pref width you want after scaling, so I think the only way you can get that is to extend WidgetGroup and provide the values yourself. Another way would be to override the table's `getPrefWidth`, using `super.getPrefWidth` to calculate the scaling.

**Comment by crykn** [comment]
The libGDX wiki can be edited over at https://github.com/libgdx/libgdx.github.io.

**Comment by NathanSweet** [comment]
I see the video, but I can't tell what behavior is incorrect. It's hard to say from your description, but one thing that sounds fishy is setting the pref size to the current size. Generally the pref size shouldn't depend on the current size. If the layout uses the pref size to size a child, what is the value the first time? It's more likely the pref size should be the largest size that makes sense, as that's the size it wants in the ideal case. When there's not enough room the layout can size it down to the min size. If there is more room than the pref size, it can size it up to the max size. If it isn't useful to size larger than the pref size, setting the pref and max sizes the same makes sense.
